### PR TITLE
feat: Adds Okta logout to oidc-middleware 

### PR DESCRIPTION
**Pull request by swiftone**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our [guidelines](/okta/okta-oidc-js/blob/master/CONTRIBUTING.md#commit)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Adding Tests
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/okta/okta-oidc-js/issues/162

## What is the new behavior?

Adds an automatically generated /logout route that will perform token revokes and redirect to okta logout, and an automatically generated /logout/callback route to destroy the local session once the okta logout has completed

## Does this PR introduce a breaking change?
- [X] Yes
- [ ] No

If a user has an existing POST-based /logout route, the new one will take precedence.

See the **Upgrading** for migration information

## Reviewers
@robertdamphousse-okta @manueltanzi-okta

### CODE REVIEW

File: packages/oidc-middleware/src/connectUtil.js [hunk 1]
```diff
@@ -106,3 +107,17 @@ connectUtil.createCallbackHandler = context => {
     passport.authenticate('oidc')(req, res, nextHandler);
   }
 };
+
+connectUtil.createLogoutHandler = context => logout.forceLogoutAndRevoke(context);
+
+connectUtil.createLogoutCallbackHandler = context => { 
+  return (req, res, next) => { 
+    if( req.session[context.options.sessionKey].state !== req.query.state ) {
```
Comment (manueltanzi-okta) [inline packages/oidc-middleware/src/connectUtil.js]: nit: space between `if` and `(`

File: packages/oidc-middleware/src/ExpressOIDC.js [hunk 2]
```diff
@@ -68,28 +79,51 @@ module.exports = class ExpressOIDC extends EventEmitter {
...
+      sessionKey: sessionKey || `oidc:${issuer}`,
       maxClockSkew: 120
-    }, options)
+    }, options);
+
+    // Build redirect uris unless they were explicitly set
+    // DEPRECATED: `redirect_uri` is deprecated in favor of `loginRedirectUri` (and most users will just set `appBaseUrl` and not set either of these)
+    // DEPRECATED: `options.routes.callback` is deprecated in favor of `options.routes.loginCallback`
+    // Fallbacks for deprecated options will be removed in a future update
+    options.loginRedirectUri = loginRedirectUri || redirect_uri || `${appBaseUrl}${options.routes.loginCallback.path}` || `${appBaseUrl}${options.routes.callback.path}`;
```
Comment (manueltanzi-okta) [inline packages/oidc-middleware/src/ExpressOIDC.js]: nit: maybe worth to put option.routes in a const too

File: packages/oidc-middleware/src/oidcUtil.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ oidcUtil.createClient = context => {
     issuer,
     client_id,
     client_secret,
-    redirect_uri,
+    loginRedirectUri: redirect_uri,
```
Comment (manueltanzi-okta) [inline packages/oidc-middleware/src/oidcUtil.js]: Just to double check, here we need `loginRedirectUri` but not `logoutRedirectUri`?

File: packages/oidc-middleware/README.md [hunk 4]
```diff
@@ -180,11 +192,11 @@ oidc.on('ready', () => {
 
 #### oidc.on('error', callback)
 
-This is triggered if an error occurs while ExpressOIDC is trying to start.
+This is triggered if an error occurs while ExpressOIDC is trying to start or if an error occurs while calling the Okta /revoke service on the users tokens while logging out.
```
[the remark this answers is not in the document]
Comment (swiftone) [reply packages/oidc-middleware/README.md]: That's deliberate - we're calling the /revoke service endpoint.  Suggestions as to a way to make that more clear?

File: packages/oidc-middleware/src/ExpressOIDC.js (hunk 2 above)
Comment (swiftone) [reply packages/oidc-middleware/src/ExpressOIDC.js]: Good point.

File: packages/oidc-middleware/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@okta/oidc-middleware",
-  "version": "1.0.2",
+  "version": "2.0.0",
```
[the remark this answers is not in the document]
Comment (swiftone) [reply packages/oidc-middleware/package.json]: Per outside discussion, agreed - I've removed the backwards compatiblity

**Pull request closed without merging**